packages/ember-metal/lib/map.js

```diff
@@ -271,7 +271,7 @@ class Map {
     let guid = guidFor(key);
 
     // ensure we don't store -0
-    let k = key === -0 ? 0 : key;
+    let k = key === -0 ? 0 : key; // eslint-disable-line no-compare-neg-zero
```

Comment:
There are explicit tests around negative zero, which force me to disable the rule for this line. In the long run we will be deprecating and removing all of this code.